scripts/asm-installer/vsphere/create_attach_context

```diff
@@ -0,0 +1,196 @@
+#!/bin/bash
+
+#set -ex
+set -CeE
+set -o pipefail
+set +o noclobber
+
+main() {
+  echo "============================================="
+  init
+  parse_validate_args "${@}"
+}
+
+init() {
+  # BSD-style readlink apparently doesn't have the same -f toggle on readlink
+  case "$(uname)" in
+  Linux) APATH="readlink" ;;
+  Darwin) APATH="stat" ;;
+  *) ;;
+  esac
+}
+
+parse_validate_args() {
+  # shellcheck disable=SC2064
+  trap "$(shopt -p nocasematch)" RETURN
+  shopt -s nocasematch
+
+  CONTEXT_FILE=".${1}"
+  read_args "${@}"
+  fill_unspecified_context
+  write_config_context
+
+}
+
+fill_unspecified_context() {
+
+  if [[ -z "${KUBECONFIG}" ]]; then
+    KUBECONFIG="$HOME/.kube/config"
+  fi
+  echo "Using KUBECONFIG ${KUBECONFIG}"
+
+  # use the absolute path for the key file
+  if [[ -f "${KEY_FILE}" ]]; then
+    KEY_FILE="$(apath -f "${KEY_FILE}")"
+    #    readonly KEY_FILE
+  elif [[ -n "${KEY_FILE}" ]]; then
+    fatal "Couldn't find key file ${KEY_FILE}. "
+  fi
+
+  CLUSTER_NAME=$(kubectl -n istio-system get IstioOperator installed-state --output=json | jq -r '.spec.values.global.multiCluster.clusterName')
+  if [[ ${CLUSTER_NAME} != ${CLUSTER} ]]; then
```

Comment:
In single cluster deployment, do we need cluster name?

Reply:
No need.